src/Rector/Expr/ModelComparisonToIsMethodRector.php

```diff
@@ -0,0 +1,150 @@
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Convert model ID comparisons to use the is() method',
+            [new CodeSample(
+                <<<'CODE_SAMPLE'
+$team->user_id === $user->id;
+$post->author_id === $author->id;
+CODE_SAMPLE,
+                <<<'CODE_SAMPLE'
+$team->user()->is($user);
+$post->author()->is($author);
+CODE_SAMPLE
+            )]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Equal::class, Identical::class];
+    }
+
+    /**
+     * @param  Equal|Identical  $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $node instanceof Equal && ! $node instanceof Identical) {
+            return null;
+        }
+
+        $result = $this->matchModelComparison($node);
+        if ($result === null) {
+            return null;
+        }
+
+        [$leftVar, $relationshipName, $rightVar] = $result;
+
+        if (! $this->couldBeModel($leftVar) || ! $this->couldBeModel($rightVar)) {
+            return null;
+        }
+
+        $methodCall = new MethodCall($leftVar, new Identifier($relationshipName));
+
+        return new MethodCall($methodCall, new Identifier('is'), [new Arg($rightVar)]);
+    }
+
+    /**
+     * @return array{Expr, string, Expr}|null
+     */
+    private function matchModelComparison(Equal|Identical $node): ?array
+    {
+        $left = $node->left;
+        $right = $node->right;
+
+        if (! $left instanceof PropertyFetch || ! $right instanceof PropertyFetch) {
+            return null;
+        }
+
+        $leftProperty = $left->name;
+        $rightProperty = $right->name;
+
+        if (! $leftProperty instanceof Identifier || ! $rightProperty instanceof Identifier) {
+            return null;
+        }
+
+        $leftPropertyName = $leftProperty->name;
+        $rightPropertyName = $rightProperty->name;
+
+        if ($this->isForeignKeyToIdPattern($leftPropertyName, $rightPropertyName)) {
+            // $model->foreign_key_id == $otherModel->id
+            $relationshipName = $this->extractRelationshipName($leftPropertyName);
+
+            return [$left->var, $relationshipName, $right->var];
+        }
+
+        if ($this->isForeignKeyToIdPattern($rightPropertyName, $leftPropertyName)) {
+            // $otherModel->id == $model->foreign_key_id
+            $relationshipName = $this->extractRelationshipName($rightPropertyName);
+
+            return [$right->var, $relationshipName, $left->var];
+        }
+
+        return null;
+    }
+
+    private function isForeignKeyToIdPattern(string $leftProperty, string $rightProperty): bool
+    {
+        return str_ends_with($leftProperty, '_id') && $rightProperty === 'id';
+    }
+
+    private function extractRelationshipName(string $foreignKeyProperty): string
+    {
+        return substr($foreignKeyProperty, 0, -3);
+    }
```

Comment:
I'm afraid this part here won't work well with a lot of cases.

It assumes that the key is always 'id' or follows the standard '_id'. I'm not sure if we can find the key name from the model's `getKeyName()` method or the `$primaryKey` property.

Also, it fails when relationship names are complex (`nationalTeam()` with `national_team_id` key), or don't exist (`external_team_id`, but no `externalTeam()` relationship).

The only way this could work well would be to check the relationships.